report/delayedusers/classes/report.php

```diff
@@ -127,15 +127,20 @@ public function fetch_data() {
      * @return array($sql, $whereparams);
      */
     public function get_submissions_sql() {
+        $whereparams = array();
         $submissiontable = 'SELECT userid, surveyproid
                             FROM {surveypro_submission}
-                            GROUP BY userid';
+                            WHERE surveyproid = :subsurveyproid
+                            GROUP BY userid, surveyproid';
+        $whereparams['subsurveyproid'] = $this->surveypro->id;
         $sql = 'SELECT '.user_picture::fields('u').', s.surveyproid
                 FROM {user} u
-                JOIN ('.$submissiontable.') s ON s.userid = u.id';
+                LEFT JOIN ('.$submissiontable.') s ON s.userid = u.id';
 
-        list($middlesql, $whereparams) = $this->get_middle_sql();
+        $actualrelation = ($this->groupid == 0) ? true : false;
```

Comment:
I'm not sure if I understand this... 

1) why are you setting `$whereparams['subsurveyproid']`both in `get_submissions_sql()`and `get_middle_sql()`?
2) in which case does it have sense to be null? Reports are always about a given survey, isn't it?
3) You are calculating `$actualrelation`but not passing it to `get_middle_sql()`is that correct?

Something does not fit in my brain with this... ciao :-)